api/v1alpha1/conditions.go

```diff
@@ -15,3 +15,25 @@ type StardogCondition struct {
 }
 
 type StardogConditionType string
+
+const (
+	// StardogReady tracks if the Stardog has been successfully reconciled.
+	StardogReady StardogConditionType = "Ready"
+	// StardogErrored is given when no connection could be established with the Stardog Database
+	// any other reconciliation error.
+	StardogErrored StardogConditionType = "Errored"
+	// StardogInvalid is given when the the Stardog Spec contains invalid properties. The specification will not be
+	// reconciled.
+	StardogInvalid StardogConditionType = "Invalid"
+	// StardogToBeDeleted is given when the the Stardog Instance has to be deleted. The reconciliation will run until
+	// the Specification is deleted.
+	StardogToBeDeleted StardogConditionType = "ToBeDeleted"
```

Comment:
This condition should only be active when the deletion does not work at first attempt. If it works in the first attempt, why should we add a condition when it's about to be deleted from etcd? So, I'm not sure if this condition is even needed or if we can use the `StardogErrored` condition for the same purpose. If a dedicated condition is needed, at least I propose to use known names (like Namespaces, or Pods).
```suggestion
	// Terminating is given when the the Stardog resource is to be deleted but the object's finalizers cannot be cleared for a reason.
	Terminating StardogConditionType = "Terminating"
```

Reply:
The naming was wrong :) The logic is actually the way you described. I will updated it to "Terminating"